tools/onenote-migration/test_migrate.py

```
@@ -0,0 +1,300 @@
+        self.assertEqual(M.retry_with_backoff(fn, "url", retries=3, base=0).status_code, 200)
+        self.assertEqual(fn.call_count, 3)
+    @patch("onenote_migrate.time.sleep")
+    def test_retries_on_503(self, mock_sleep):
+        ok = self._ok(200); err = self._ok(503); err.headers = {}
+        fn = MagicMock(side_effect=[err, ok])
+        self.assertEqual(M.retry_with_backoff(fn, "url", retries=3, base=0).status_code, 200)
+    @patch("onenote_migrate.time.sleep")
+    def test_raises_after_max_retries(self, mock_sleep):
+        throttled = self._ok(429); throttled.headers = {"Retry-After": "0"}
+        throttled.raise_for_status = MagicMock(side_effect=Exception("too many"))
+        with self.assertRaises(Exception):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)